### PR TITLE
Allow to disable decoding/encoding path in uri in URL engine

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add new setting `disable_url_encoding` that allows to disable decoding/encoding path in uri in URL engine.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by Avogar** [comment]
Usecase:
```
avogar-dev :) select count() FROM url('https://huggingface.co/datasets/blog_authorship_corpus/resolve/refs%2Fconvert%2Fparquet/blog_authorship_corpus/blog_authorship_corpus-train-00000-of-00002.parquet') SETTINGS max_http_get_redirects=10, decode_and_encode_path_in_url=1

SELECT count()
FROM url('https://huggingface.co/datasets/blog_authorship_corpus/resolve/refs%2Fconvert%2Fparquet/blog_authorship_corpus/blog_authorship_corpus-train-00000-of-00002.parquet')
SETTINGS max_http_get_redirects = 10, decode_and_encode_path_in_url = 1

Query id: 37e9ddc8-bb33-4f20-a665-d8db7fdf0ae8


0 rows in set. Elapsed: 1.307 sec.

Received exception:
Code: 86. DB::HTTPException: Received error from remote server /datasets/blog_authorship_corpus/resolve/refs/convert/parquet/blog_authorship_corpus/blog_authorship_corpus-train-00000-of-00002.parquet. HTTP status code: 404 Not Found, body: Invalid rev id: refs: Cannot extract table structure from Parquet format file. You can specify the structure manually. (RECEIVED_ERROR_FROM_REMOTE_IO_SERVER)

avogar-dev :) select count() FROM url('https://huggingface.co/datasets/blog_authorship_corpus/resolve/refs%2Fconvert%2Fparquet/blog_authorship_corpus/blog_authorship_corpus-train-00000-of-00002.parquet') SETTINGS max_http_get_redirects=10, decode_and_encode_path_in_url=0

SELECT count()
FROM url('https://huggingface.co/datasets/blog_authorship_corpus/resolve/refs%2Fconvert%2Fparquet/blog_authorship_corpus/blog_authorship_corpus-train-00000-of-00002.parquet')
SETTINGS max_http_get_redirects = 10, decode_and_encode_path_in_url = 0

Query id: e13bdf44-e4a7-409a-90e0-1397bb3c381d

┌─count()─┐
│  456000 │
└─────────┘

1 row in set. Elapsed: 2.433 sec. Processed 456.00 thousand rows, 2.28 MB (187.39 thousand rows/s., 936.96 KB/s.)

```

**Comment by robot-ch-test-poll2** [comment]
This is an automated comment for commit 483ddb53ebfa01c02deda76a39bc44cc08df4f00 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/52337/483ddb53ebfa01c02deda76a39bc44cc08df4f00/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by pufit — APPROVED** [review]
`decode_and_encode_path` doesn't look like a clear name to me. 
Maybe `disable_url_encoding'? Not insisting on it, though.

File: docs/en/sql-reference/table-functions/url.md [hunk 1]
```diff
@@ -56,7 +56,8 @@ Character `|` inside patterns is used to specify failover addresses. They are it
 ## Storage Settings {#storage-settings}
 
 - [engine_url_skip_empty_files](/docs/en/operations/settings/settings.md#engine_url_skip_empty_files) - allows to skip empty files while reading. Disabled by default.
+- [decode_and_encode_path_in_url](/docs/en/operations/settings/settings.md#decode_and_encode_path_in_url) - enables or disables decoding/encoding path in uri. Enabled by default.
 
-**See Also**
+- **See Also**
```
[the remark this answers is not in the document]
Comment (Avogar) [reply docs/en/sql-reference/table-functions/url.md:61]: Yep, will fix

File: base/poco/Foundation/src/URI.cpp [hunk 2]
```diff
@@ -681,6 +688,13 @@ void URI::decode(const std::string& str, std::string& decodedStr, bool plusAsSpa
 	}
 }
 
+void URI::encodePath(std::string & encodedStr) const
```
[the remark this answers is not in the document]
Comment (Avogar) [reply base/poco/Foundation/src/URI.cpp:691]: Agree, it will be better

**Comment by Avogar** [comment]
> decode_and_encode_path doesn't look like a clear name to me.
Maybe `disable_url_encoding'? Not insisting on it, though.

`disable_url_encoding` is better. I am not good at naming :)

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**